plugin.video.embycon/resources/lib/datamanager.py

```diff
@@ -0,0 +1,163 @@
+# Gnu General Public License - see LICENSE.TXT
+
+import hashlib
+import os
+import threading
+import json
+
+import xbmcaddon
+import xbmc
+
+from downloadutils import DownloadUtils
+from simple_logging import SimpleLogging
+from utils import getChecksum
+from kodi_utils import HomeWindow
+
+log = SimpleLogging(__name__)
+
+
+class DataManager():
+    cacheDataResult = None
+    dataUrl = None
+    cacheDataPath = None
+    canRefreshNow = False
+
+    def __init__(self, *args):
+        log.debug("DataManager __init__")
+
+    def getCacheValidatorFromData(self, result):
+        key = 'Items'
+        results = result.get(key)
+        if results is None:
+            key = 'SearchHints'
+            results = result.get(key)
+            if results is None:
+                results = []
+
+        itemCount = 0
+        dataHashString = ""
+
+        for item in results:
+            item_hash_string = getChecksum(item)
+            item_hash_string = str(itemCount) + "_" + key + "_" + item.get("Name", "-") + "_" + item_hash_string + "|"
+            log.debug("ITEM_HASH: " + item_hash_string)
+            dataHashString += item_hash_string
+            itemCount += 1
+
+        # hash the data
+        dataHashString = dataHashString.encode("UTF-8")
+        m = hashlib.md5()
+        m.update(dataHashString)
+        validatorString = m.hexdigest()
+
+        log.debug("getCacheValidatorFromData : RawData  : " + dataHashString)
+        log.debug("getCacheValidatorFromData : hashData : " + validatorString)
+
+        return validatorString
+
+    def loadJasonData(self, jsonData):
```

Comment:
Just because I noticed while reviewing your `open()` statements, I'd add some error-handling here unless you want a possible parsing exception to be thrown. You know your source better than I do :)

This is **not** a required change (github just doesn't let me write a normal comment while being in review mode :/)

```python
# Just an example
def loadJasonData(self, jsonData):
    try:
        return json.loads(jsonData)
    except json.decoder.JSONDecodeError:
        return dict()
```